### PR TITLE
feat(discover): Allow measurements in Pxx functions

### PR DESCRIPTION
**Pull request by Zylphrex**
The predefined Pxx functions is currently hard coded for transaction.duration
but there is no reason why it can't work for any numeric column. This change
adds support for any numeric column in these predefined Pxx functions while
leaving the default as transaction.duration for backwards compatibility.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

**Review by wmak — CHANGES_REQUESTED** [review]
I think this breaks when you try `p75(measurements.lcp)` cause of `validate_argument_count` not taking the default into the total number of arguments

Can you add some tests for this too?

**Comment by Zylphrex** [comment]
> I think this breaks when you try `p75(measurements.lcp)` cause of `validate_argument_count` not taking the default into the total number of arguments
> 
> Can you add some tests for this too?

No sure what you mean by `p75(measurements.lcp)` breaking as `validate_argument_count` does check against the total number of arguments, but I added some tests for it.

**Review by wmak — APPROVED** [review]
🤔 not... sure why this was failing when I tested it, nevermind looks good now

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -1561,25 +1576,34 @@ def validate_argument_count(self, field, arguments):
         ),
         Function(
             "p50",
-            aggregate=[u"quantile(0.5)", "transaction.duration", None],
+            optional_args=[with_default("transaction.duration")(NumericColumnNoLookup("column"))],
```
Comment (markstory) [inline src/sentry/api/event_search.py:1579]: I like the with_default() function but think the currying is a bit elaborate. Wouldn't `with_default('transaction.duration', NumericColumnNoLookup('column'))` work just as well?

**Review by markstory — APPROVED** [review]

**Pull request merged**